raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -480,12 +481,18 @@ private void becomeFollower(OptionalInt leaderId, int epoch) throws IOException
         }
     }
 
-    private VoteResponseData buildVoteResponse(Errors error, boolean voteGranted) {
-        return new VoteResponseData()
-                .setErrorCode(error.code())
-                .setLeaderEpoch(quorum.epoch())
-                .setLeaderId(quorum.leaderIdOrNil())
-                .setVoteGranted(voteGranted);
+    private VoteResponseData buildVoteResponse(Errors topLevelError, boolean voteGranted) {
+        return buildVoteResponse(topLevelError, Errors.NONE, voteGranted);
+    }
+
+    private VoteResponseData buildVoteResponse(Errors topLevelError, Errors partitionLevelError, boolean voteGranted) {
```

Comment:
Seems all usages of this method have `NONE` as the top-level error. Maybe we do not need the extra argument?